lib/rules/no-restricted-class.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * @fileoverview Forbid certain classes from being used
+ * @author Tao Bojlen
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+/**
+ * Report a forbidden class
+ * @param {string} className
+ * @param {*} node
+ * @param {RuleContext} context
+ * @param {Set<string>} forbiddenClasses
+ */
+const reportForbiddenClass = (className, node, context, forbiddenClasses) => {
+  if (forbiddenClasses.has(className)) {
+    const loc = node.value ? node.value.loc : node.loc
+    context.report({
+      node,
+      loc,
+      messageId: 'forbiddenClass',
+      data: {
+        class: className
+      }
+    })
+  }
+}
+
+/**
+ * @param {Expression} node
+ * @param {boolean} [textOnly]
+ * @returns {IterableIterator<{ className:string, reportNode: ESNode }>}
+ */
+function* extractClassNames(node, textOnly) {
+  if (node.type === 'Literal') {
+    yield* `${node.value}`
+      .split(/\s+/)
+      .map((className) => ({ className, reportNode: node }))
+    return
+  }
+  if (node.type === 'TemplateLiteral') {
+    for (const templateElement of node.quasis) {
+      yield* templateElement.value.cooked
+        .split(/\s+/)
+        .map((className) => ({ className, reportNode: templateElement }))
+    }
+    for (const expr of node.expressions) {
+      yield* extractClassNames(expr, true)
+    }
+    return
+  }
+  if (node.type === 'BinaryExpression') {
+    if (node.operator !== '+') {
+      return
+    }
+    yield* extractClassNames(node.left, true)
+    yield* extractClassNames(node.right, true)
+    return
+  }
```

Comment:
I made a mistake in the if statement. We need to add the following:

```suggestion
  }
  if (textOnly) {
    return
  }
```

`<template><div :class="''+{forbidden:true}">Content</div></template>` is actually drawn as `<div class="[object Object]">Content</div>`.

https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPjxkaXYgOmNsYXNzPVwiJycre2ZvcmJpZGRlbjp0cnVlfVwiPkNvbnRlbnQ8L2Rpdj48L3RlbXBsYXRlPiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vc2ZjLnZ1ZWpzLm9yZy92dWUucnVudGltZS5lc20tYnJvd3Nlci5qc1wiXG4gIH1cbn0ifQ==

Reply:
Thank you; fixed -- and added a test case.